### DISCUSSION TITLE
Unable to delete Line

### DISCUSSION THREAD

**Discussion by lyflyflxf** [mozman/ezdxf#355] [Q&A] [answered]
I'm using python 3.6 and ezdxf 0.15.1 processing a DXF file generated by UG NX 12.0 under windows 7 and windows 10.

I tried to delete all LINE objects, with code
`doc = ezdxf.readfile(path)
msp = doc.modelspace()
for entity in msp:
    if entity.dxftype().lower()=='line':
        msp.delete_entity(entity)

for entity in msp:
    print(entity.dxftype())`

Then with print, I found out that some objects are still LINEs.  I wouldn't have noticed this if doc.save() had not raised the exception "AttributeError: 'Line' object has no attribute 'dxf' "

Why did this happen?

**Comment by mozman** [comment, marked as answer]
Most likely, the model space contains block references (INSERT) that contain LINE entities.

PS: upgrade your Python version, the next ezdxf version requires at least Python 3.7

**Reply by lyflyflxf** [reply]
Thanks.  I solved this by avoiding producing blocks in the DXF, because I failed to delete all blocks by deleting all INSERT entities.  Besides, some LINE entities cannot be deleted either.  What can I do to delete them?

P.S. I will upgrade my Python later.
